<!DOCTYPE html>
<html lang='en'>
<head>
<title>0011-exportfs-only-do-glibc-specific-hackery-on-glibc.patch\nfs-utils\main - aports - Main aports tree
</title>
<meta name='generator' content='cgit v1.0'/>
<meta name='robots' content='index, nofollow'/>
<link rel='stylesheet' type='text/css' href='/cgit.css'/>
<link rel='shortcut icon' href='//alpinelinux.org/alpine-logo.ico'/>
<link rel='alternate' title='Atom feed' href='https://git.alpinelinux.org/cgit/aports/atom/main/nfs-utils/0011-exportfs-only-do-glibc-specific-hackery-on-glibc.patch?h=master' type='application/atom+xml'/>
<link rel='vcs-git' href='git://git.alpinelinux.org/aports' title='aports Git repository'/>
<link rel='vcs-git' href='https://git.alpinelinux.org/cgit/aports' title='aports Git repository'/>
</head>
<body>
<div id='cgit'><table id='header'>
<tr>
<td class='logo' rowspan='2'><a href='/cgit/'><img src='//wiki.alpinelinux.org/w/resources/assets/alogo.png' alt='cgit logo'/></a></td>
<td class='main'><a href='/cgit/'>index</a> : <a title='aports' href='/cgit/aports/'>aports</a></td><td class='form'><form method='get'>
<select name='h' onchange='this.form.submit();'>
<option value='1.10-stable'>1.10-stable</option>
<option value='1.9'>1.9</option>
<option value='2.0-stable'>2.0-stable</option>
<option value='2.1-stable'>2.1-stable</option>
<option value='2.2-stable'>2.2-stable</option>
<option value='2.3-stable'>2.3-stable</option>
<option value='2.4-stable'>2.4-stable</option>
<option value='2.5-stable'>2.5-stable</option>
<option value='2.6-stable'>2.6-stable</option>
<option value='2.7-stable'>2.7-stable</option>
<option value='3.0-stable'>3.0-stable</option>
<option value='3.1-stable'>3.1-stable</option>
<option value='3.2-stable'>3.2-stable</option>
<option value='3.3-stable'>3.3-stable</option>
<option value='3.4-stable'>3.4-stable</option>
<option value='3.5-stable'>3.5-stable</option>
<option value='3.6-stable'>3.6-stable</option>
<option value='3.7-stable'>3.7-stable</option>
<option value='master' selected='selected'>master</option>
</select> <input type='submit' value='switch'/></form></td></tr>
<tr><td class='sub'>Main aports tree
</td><td class='sub right'>gitolite</td></tr></table>
<table class='tabs'><tr><td>
<a href='/cgit/aports/about/'>about</a><a href='/cgit/aports/'>summary</a><a href='/cgit/aports/refs/'>refs</a><a href='/cgit/aports/log/main/nfs-utils/0011-exportfs-only-do-glibc-specific-hackery-on-glibc.patch'>log</a><a class='active' href='/cgit/aports/tree/main/nfs-utils/0011-exportfs-only-do-glibc-specific-hackery-on-glibc.patch'>tree</a><a href='/cgit/aports/commit/main/nfs-utils/0011-exportfs-only-do-glibc-specific-hackery-on-glibc.patch'>commit</a><a href='/cgit/aports/diff/main/nfs-utils/0011-exportfs-only-do-glibc-specific-hackery-on-glibc.patch'>diff</a><a href='/cgit/aports/stats/main/nfs-utils/0011-exportfs-only-do-glibc-specific-hackery-on-glibc.patch'>stats</a></td><td class='form'><form class='right' method='get' action='/cgit/aports/log/main/nfs-utils/0011-exportfs-only-do-glibc-specific-hackery-on-glibc.patch'>
<select name='qt'>
<option value='grep'>log msg</option>
<option value='author'>author</option>
<option value='committer'>committer</option>
<option value='range'>range</option>
</select>
<input class='txt' type='text' size='10' name='q' value=''/>
<input type='submit' value='search'/>
</form>
</td></tr></table>
<div class='path'>path: <a href='/cgit/aports/tree/'>root</a>/<a href='/cgit/aports/tree/main'>main</a>/<a href='/cgit/aports/tree/main/nfs-utils'>nfs-utils</a>/<a href='/cgit/aports/tree/main/nfs-utils/0011-exportfs-only-do-glibc-specific-hackery-on-glibc.patch'>0011-exportfs-only-do-glibc-specific-hackery-on-glibc.patch</a></div><div class='content'>blob: 2d4f61311461ba2fb4642a9b4697661dbaf2c01a (<a href='/cgit/aports/plain/main/nfs-utils/0011-exportfs-only-do-glibc-specific-hackery-on-glibc.patch'>plain</a>)
<table summary='blob content' class='blob'>
<tr><td class='linenumbers'><pre><a id='n1' href='#n1'>1</a>
<a id='n2' href='#n2'>2</a>
<a id='n3' href='#n3'>3</a>
<a id='n4' href='#n4'>4</a>
<a id='n5' href='#n5'>5</a>
<a id='n6' href='#n6'>6</a>
<a id='n7' href='#n7'>7</a>
<a id='n8' href='#n8'>8</a>
<a id='n9' href='#n9'>9</a>
<a id='n10' href='#n10'>10</a>
<a id='n11' href='#n11'>11</a>
<a id='n12' href='#n12'>12</a>
<a id='n13' href='#n13'>13</a>
<a id='n14' href='#n14'>14</a>
<a id='n15' href='#n15'>15</a>
<a id='n16' href='#n16'>16</a>
<a id='n17' href='#n17'>17</a>
<a id='n18' href='#n18'>18</a>
<a id='n19' href='#n19'>19</a>
<a id='n20' href='#n20'>20</a>
<a id='n21' href='#n21'>21</a>
<a id='n22' href='#n22'>22</a>
<a id='n23' href='#n23'>23</a>
<a id='n24' href='#n24'>24</a>
<a id='n25' href='#n25'>25</a>
<a id='n26' href='#n26'>26</a>
<a id='n27' href='#n27'>27</a>
<a id='n28' href='#n28'>28</a>
<a id='n29' href='#n29'>29</a>
<a id='n30' href='#n30'>30</a>
<a id='n31' href='#n31'>31</a>
<a id='n32' href='#n32'>32</a>
<a id='n33' href='#n33'>33</a>
<a id='n34' href='#n34'>34</a>
<a id='n35' href='#n35'>35</a>
<a id='n36' href='#n36'>36</a>
</pre></td>
<td class='lines'><pre><code>From 0b4df13026eb371b68abc4b8cec4ca8d90c3b0a3 Mon Sep 17 00:00:00 2001
From: Natanael Copa &lt;ncopa&#64;alpinelinux.org&gt;
Date: Mon, 2 Jun 2014 14:42:10 +0200
Subject: [PATCH] exportfs: only do glibc specific hackery on glibc

We should not depend on the libc do free(3) on ai_canonname as that is
completely up to implementation and known o break things on uclibc and
musl libc.

Signed-off-by: Natanael Copa &lt;ncopa&#64;alpinelinux.org&gt;
<span style="color:#0057ae">---</span>
 support/export/hostname.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/support/export/hostname.c b/support/export/hostname.c
index d9153e1..30584b4 100644
<span style="color:#0057ae">--- a/support/export/hostname.c</span>
<span style="color:#000000; font-weight:bold">+++ b/support/export/hostname.c</span>
<span style="color:#010181">&#64;&#64; -384,6 +384,7 &#64;&#64;</span>
 
 	ai = host_pton(buf);
 
<span style="color:#000000; font-weight:bold">+#if !definded(__UCLIBC__) &amp;&amp; defined(__GLIBC__)</span>
 	/*
 	 * getaddrinfo(AI_NUMERICHOST) never fills in ai_canonname
 	 */
<span style="color:#010181">&#64;&#64; -394,7 +395,9 &#64;&#64;</span>
 			ai = NULL;
 		}
 	}
<span style="color:#000000; font-weight:bold">+#endif</span>
 
 	return ai;
 }
<span style="color:#000000; font-weight:bold">+</span>
 #endif	/* !HAVE_GETNAMEINFO */
</code></pre></td></tr></table>
</div> <!-- class=content -->
<script type="text/javascript">
var gaJsHost = (("https:" == document.location.protocol) ? "https://ssl." : "http://www.");
document.write(unescape("%3Cscript src='" + gaJsHost + "google-analytics.com/ga.js' type='text/javascript'%3E%3C/script%3E"));
</script>
<script type="text/javascript">
try {
var pageTracker = _gat._getTracker("UA-2804446-12");
pageTracker._trackPageview();
} catch(err) {}</script>
</div> <!-- id=cgit -->
</body>
</html>
